software/apps/romi_control/main.c

```diff
@@ -1,183 +1,308 @@
-    connected = true;
-}
-
-
-void ble_evt_write(ble_evt_t const* p_ble_evt) {
-    //logic for each characteristic and related state changes
-    //Try not to modify the state here...
-    readInput();
-}
-
-void print_state(states current_state){
-    char buf[16];
-	switch(current_state){
-	case OFF: {
-		display_write("OFF", DISPLAY_LINE_0);
-        snprintf(buf, 16, "", waypoint[1]);
-		display_write(buf, DISPLAY_LINE_1);
-		break;
-      }
-	case WAITING: {
-		display_write("WAITING", DISPLAY_LINE_0);
-        snprintf(buf, 16, "", waypoint[1]);
-		display_write(buf, DISPLAY_LINE_1);
-		break;
-    }
-	case TURNING: {
-        snprintf(buf, 16, "TURN TARGET: %f", waypoint[1]);
-        display_write(buf, DISPLAY_LINE_0);
-        snprintf(buf, 16, "ANGLE CUR: %f", lsm9ds1_read_gyro_integration().z_axis);
-		display_write(buf, DISPLAY_LINE_1);
-		break;
-    }
-	case DRIVING: {
-        snprintf(buf, 16, "TARGET: %.2f", waypoint[0]);
-		display_write(buf, DISPLAY_LINE_0);
-        snprintf(buf, 16, "L: %.2f R: %.2f", total_distance_left, total_distance_right);
-		display_write(buf, DISPLAY_LINE_1);
-		break;
-    }
-   }
-}
-
-static float measure_distance(uint16_t current_encoder, uint16_t previous_encoder) {
-  const float CONVERSION = 0.0006108;
-  float distance = 0;
-  if (current_encoder < previous_encoder) {
-      if (previous_encoder - current_encoder > 30000) {
-        distance = (current_encoder - previous_encoder + 655365);
-      }
-      else {
-        distance = current_encoder - previous_encoder;
-      }
-  }
-  else {
-      if (current_encoder  - previous_encoder > 30000) {
-        distance = (current_encoder - previous_encoder - 655365);
-      }
-      else {
-        distance = current_encoder - previous_encoder;
-      }
-  }
-  float val = CONVERSION * distance;
-  if (fabs(val) > 300) {
-     val = 0;
-  }
-  return val;
-}
 
 int main(void) {
-  ret_code_t error_code = NRF_SUCCESS;
-
-  // initialize RTT library
-  error_code = NRF_LOG_INIT(NULL);
-  APP_ERROR_CHECK(error_code);
-  NRF_LOG_DEFAULT_BACKENDS_INIT();
-  printf("Log initialized!\n");
-
-  // Setup BLE
-  simple_ble_app = simple_ble_init(&ble_config);
-
-  simple_ble_add_service(&robot_service);
-
-  //Register your characteristics
-  simple_ble_add_characteristic(0, 1, 0, 0,
-      sizeof(waypoint), (uint8_t*)&waypoint,
-      &robot_service, &waypoint_char);
-
-  simple_ble_add_characteristic(1, 0, 1, 0, 
-    sizeof(acknowledged), (uint8_t*)&acknowledged, 
-    &robot_service, &ack_char);
-  
-  // Start Advertising
-  simple_ble_adv_only_name();
-
-  // initialize LEDs
-  nrf_gpio_pin_dir_set(23, NRF_GPIO_PIN_DIR_OUTPUT);
-  nrf_gpio_pin_dir_set(24, NRF_GPIO_PIN_DIR_OUTPUT);
-  nrf_gpio_pin_dir_set(25, NRF_GPIO_PIN_DIR_OUTPUT);
-
-  nrf_gpio_pin_set(23);
-  nrf_gpio_pin_set(24);
-  nrf_gpio_pin_set(25);
-
-
+  setup();
 
   // initialize display
   nrf_drv_spi_t spi_instance = NRF_DRV_SPI_INSTANCE(1);
```

Comment:
Shouldn't this go in setup as well?

Reply:
We tried; the display wouldn't initialize. Don't know why...